tests/Extensions/Services.Remoting.UnitTests/OmexServiceRemotingClientFactoryTests.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Fabric;
+using System.IO;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Omex.Extensions.Services.Remoting.Client;
+using Microsoft.Omex.Extensions.Services.Remoting.Runtime;
+using Microsoft.ServiceFabric.Services.Client;
+using Microsoft.ServiceFabric.Services.Communication.Client;
+using Microsoft.ServiceFabric.Services.Remoting;
+using Microsoft.ServiceFabric.Services.Remoting.V2;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Client;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using ServiceFabric.Mocks;
+
+namespace Services.Remoting
+{
+	[TestClass]
+	public class OmexServiceRemotingClientFactoryTests
+	{
+		[TestMethod]
+		public void GetRemotingMessageBodyFactory_PropagatesCalls()
+		{
+			IServiceRemotingMessageBodyFactory expectedResult = new Mock<IServiceRemotingMessageBodyFactory>().Object;
+			Mock<IServiceRemotingClientFactory> factoryMock = new Mock<IServiceRemotingClientFactory>();
+			factoryMock.Setup(f => f.GetRemotingMessageBodyFactory()).Returns(expectedResult);
+			OmexServiceRemotingClientFactory wrapper = new OmexServiceRemotingClientFactory(factoryMock.Object);
+
+			IServiceRemotingMessageBodyFactory actualResult = wrapper.GetRemotingMessageBodyFactory();
+			factoryMock.Verify(f => f.GetRemotingMessageBodyFactory());
+
+			Assert.AreEqual(expectedResult, actualResult);
+		}
+
+		[TestMethod]
+		public async Task GetClientAsync_UsingUri_WrapsClient()
+		{
+			Uri serviceUri = new Uri("http://localhost");
+			ServicePartitionKey partitionKey = new ServicePartitionKey();
+			TargetReplicaSelector targetReplicaSelector = TargetReplicaSelector.PrimaryReplica;
+			string listenerName = nameof(GetClientAsync_UsingUri_WrapsClient);
+			OperationRetrySettings retrySettings = new OperationRetrySettings();
+			CancellationToken cancellationToken = CancellationToken.None;
+
+			Expression<Func<IServiceRemotingClientFactory, Task<IServiceRemotingClient>>> expression = f =>
+				f.GetClientAsync(
+					serviceUri,
+					partitionKey,
+					targetReplicaSelector,
+					listenerName,
+					retrySettings,
+					cancellationToken);
+
+			IServiceRemotingClient expectedResult = new Mock<IServiceRemotingClient>().Object;
+			Mock<IServiceRemotingClientFactory> factoryMock = new Mock<IServiceRemotingClientFactory>();
+			factoryMock.Setup(expression).Returns(Task.FromResult(expectedResult));
+			OmexServiceRemotingClientFactory wrapper = new OmexServiceRemotingClientFactory(factoryMock.Object);
+
+			IServiceRemotingClient actualResult = await wrapper.GetClientAsync(
```

Comment:
maybe ConfigureAwait

Reply:
Added, thank you, constantly missing them in unit tests